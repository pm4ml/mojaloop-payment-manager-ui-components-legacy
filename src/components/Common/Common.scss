@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/fonts.scss';

.validation__icon {
  fill: transparent;
}

.validation__icon.validation__icon--invalid {
  height: auto;
  width: auto;
  fill: $color__main-danger;
}

.validation__messages {
  list-style-type: none;
  margin: 0;
  padding: 14px;
  padding-right: 18px;
  background: $color__gray-1;
  box-shadow: 0px 1px 2px $color__alpha-2;
  border-radius: 3px;
  min-width: 60px;
  display: block;
}

.validation__message {
  font-family: $font__primary, $font__primary-fallback;
  letter-spacing: $font__primary-spacing;
  display: flex;
  align-items: center;
  font-size: 12px;
  color: $color__dark-gray-1;
  min-width: 60px;
}

.validation__message--inactive {
  text-decoration: line-through;
  stroke: none;
  color: $color__gray-5;
}

.validation__message__text {
  white-space: nowrap;
}

.validation__message__icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.validation__message__icon--active {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-danger;
  stroke: $color__main-danger;
}

.validation__message__icon--inactive {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-success;
}

.validation__message__icon--unset {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  height: 6px;
  width: 6px;
  margin: 3px;
  border-radius: 3px;
  display: flex;
  background: #ddd;
}

@keyframes fadeInValidationMessageIcon {
  0% {
    transform: scale(0.5, 0.5);
    opacity: 0.5;
  }

  100% {
    transform: scale(1, 1);
    opacity: 1;
  }
}

.named__icon {
  height: 20px;
  width: 20px;
  border-radius: 20px;
  font-size: 13px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #022843;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}
