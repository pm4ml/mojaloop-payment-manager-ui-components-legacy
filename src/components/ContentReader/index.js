import ContentReader from './ContentReader';

export default ContentReader;
