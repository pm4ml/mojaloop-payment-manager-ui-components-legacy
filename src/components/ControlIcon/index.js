import ControlIcon from './ControlIcon';

export default ControlIcon;
