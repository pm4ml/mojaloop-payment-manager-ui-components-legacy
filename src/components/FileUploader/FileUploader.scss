@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/transitions.scss';

.input-fileuploader {
}
.input-fileuploader__component {
  display: flex;
  position: relative;
  cursor: pointer;
  transition: $input__main-transition;

  &.input-fileuploader__mb-input--open {
  }

  &.input-fileuploader__mb-input--disabled {
  }
}
.input-fileuploader-box {
  display: flex;
  position: relative;
  height: 100%;
  width: 100%;
}

.input-fileuploader__placeholder {
}

.input-fileuploader__content {
}

.input-fileuploader__input {
  flex: 0 0 0px;
  position: relative;
  height: 20px;
  opacity: 0;
  width: 0px;
  min-width: 0px;
}

.input-fileuploader__filename {
  display: flex;
  align-items: center;
  flex: 2 1 auto;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  position: relative;
  font-size: 14px;
  color: $color__dark-gray-3;
  overflow: hidden;
  padding: 0px;
  padding-top: 5px;
  height: 100%;

  &.input-fileuploader__filename--no-file {
    color: $color__gray-4;
  }
}
.input-fileuploader__filename--small {
  font-size: 10px;
}
.input-fileuploader__filename--medium {
  font-size: 12px;
}
.input-fileuploader__filename--large {
  font-size: 14px;
}

.input-fileuploader__button {
}

.input-fileuploader__icon {
  flex: 0 0 auto;
}
