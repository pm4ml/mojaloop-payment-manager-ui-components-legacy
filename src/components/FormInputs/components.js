import './FormInput.scss';
import '../../icons/mule/forum-small.svg';

import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

export const FieldInfoOverlay = ({ assignRef, title, description, url, required }) => (
  <div className="forminput__field-info" ref={assignRef}>
    <div className="forminput__field-info__title-container">
      <div className="forminput__field-info__title">{title}</div>
      {required && (
        <React.Fragment>
          <div className="forminput__field-info__required-icon">
            <Icon size={20} name="info-small" className="forminput__label-icon" fill="#333" />
          </div>
          <div className="forminput__field-info__required-label"> required </div>
        </React.Fragment>
      )}
    </div>
    <div className="forminput__field-info__content">
      <p>{description}</p>
    </div>
    {url && (
      <div className="forminput__field-info__url">
        <a href={url}>{url}</a>
      </div>
    )}
  </div>
);

export class FieldInfo extends PureComponent {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
    this.closeIfClickingOutside = this.closeIfClickingOutside.bind(this);
    this.state = {
      open: false,
    };
  }
  componentWillUnmount() {
    this.close();
  }
  onClick() {
    if (this.state.open) {
      this.close();
    } else {
      this.open();
    }
  }
  open() {
    this.area = React.createRef();
    this.setState(
      {
        open: true,
      },
      () => {
        document.addEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  close() {
    this.setState(
      {
        open: false,
      },
      () => {
        this.area = null;
        document.removeEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  closeIfClickingOutside(evt) {
    if (!this.area.current) {
      return;
    }
    if (!this.area.current.contains(evt.target)) {
      this.close();
    }
  }
  render() {
    const { iconSize, title, required, description, url } = this.props;
    let content = null;
    if (this.state.open) {
      content = (
        <FieldInfoOverlay
          assignRef={this.area}
          title={title}
          description={description}
          url={url}
          required={required}
        />
      );
    }
    const fieldInfoClassName = utils.composeClassNames([
      'forminput__field-info__icon',
      this.state.open && 'forminput__field-info__icon--active',
    ]);

    return (
      <Tooltip
        forceVisibility={this.state.open}
        showOnHover={false}
        content={content}
        custom
        position="right"
        align="start"
      >
        <Icon
          name="forum-small"
          onClick={this.onClick}
          size={iconSize}
          className={fieldInfoClassName}
        />
      </Tooltip>
    );
  }
}

export const Label = ({ size, label, required, complete, description, url }) => {
  if (!label) {
    return null;
  }

  const className = utils.composeClassNames([
    'forminput__label-box',
    size === 's' && 'forminput__label-box--small',
    size === 'm' && 'forminput__label-box--medium',
    size === 'l' && 'forminput__label-box--large',
  ]);
  const iconSizes = {
    s: 11,
    m: 12,
    l: 14,
  };
  const iconSize = iconSizes[size];
  return (
    <div className={className}>
      {required && (
        <Icon
          size={iconSize}
          name="info-small"
          className="forminput__label-icon"
          fill={complete ? '#39f' : '#f93'}
          tooltip={complete ? '' : 'This is a required field'}
        />
      )}
      <label>{label}</label>
      {description && (
        <FieldInfo
          iconSize={iconSize}
          title={label}
          description={description}
          url={url}
          required={required}
        />
      )}
    </div>
  );
};

export const InlineButton = ({ visible, isLocked, isDisabled, onClick, label }) => {
  if (!visible || isLocked || isDisabled) {
    return null;
  }

  return (
    <Button
      size="m"
      noFill
      kind="secondary"
      className="forminput__inline-button"
      onClick={onClick}
      disabled={isDisabled}
      label={label}
    />
  );
};

export const LockedIcon = ({ locked }) => {
  if (!locked) {
    return null;
  }
  return (
    <div className="forminput__inline-icon">
      <Icon
        name="lock-small"
        size={20}
        fill="#999"
        style={{ marginLeft: '10px' }}
        tooltip="this field is locked"
      />
    </div>
  );
};

export const InfoMessage = ({ message }) => (
  <div className="forminput__input-message">{message && <Tooltip>{message}</Tooltip>}</div>
);
