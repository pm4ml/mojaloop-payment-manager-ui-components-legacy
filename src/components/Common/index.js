import './Common.scss';
import '../../icons/mule/check-small.svg';
import '../../icons/mule/close-small.svg';

import PropTypes from 'prop-types';
import React from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon, { iconSizes } from '../Icon';
import Row from '../Row';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

const Loader = ({ size }) => (
  <div className="mb-input__inner-icon mb-loader">
    <Spinner size={iconSizes[size]} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.oneOf(['s', 'm', 'l']),
};
Loader.defaultProps = {
  size: 'l',
};

const Placeholder = ({ label, size, active }) => {
  // The Placeholder that renders inside an input

  const placeholderClassName = utils.composeClassNames([
    'mb-input__placeholder',
    size === 's' && 'mb-input__placeholder--small',
    size === 'm' && 'mb-input__placeholder--medium',
    size === 'l' && 'mb-input__placeholder--large',
    size === 's' && active && 'mb-input__placeholder--active-small',
    size === 'm' && active && 'mb-input__placeholder--active-medium',
    size === 'l' && active && 'mb-input__placeholder--active-large',
  ]);

  return <label className={placeholderClassName}>{label}</label>;
};

Placeholder.propTypes = {
  label: PropTypes.string,
  active: PropTypes.bool,
  size: PropTypes.oneOf(['s', 'm', 'l']),
};

Placeholder.defaultProps = {
  label: undefined,
  active: false,
  size: 'l',
};

const InnerButton = ({ className, size, kind, onClick, label, disabled, noFill, icon }) => {
  // Internal button used by inputs

  let innerButtonSize;
  if (size === 'l') {
    innerButtonSize = 'm';
  } else if (size === 'm') {
    innerButtonSize = 's';
  } else {
    innerButtonSize = 'xs';
  }

  const innerButtonClassName = utils.composeClassNames([
    className,
    'mb-input__inner-button',
    size === 's' && 'mb-input__inner-button--small',
    size === 'm' && 'mb-input__inner-button--medium',
    size === 'l' && 'mb-input__inner-button--large',
  ]);
  return (
    <Button
      kind={kind}
      size={innerButtonSize}
      className={innerButtonClassName}
      icon={icon}
      noFill={noFill}
      onClick={onClick}
      tabIndex="-1"
      label={label}
      disabled={disabled}
    />
  );
};

InnerButton.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  kind: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
};

InnerButton.defaultProps = {
  className: undefined,
  size: 'l',
  kind: 'primary',
  icon: undefined,
  onClick: undefined,
  label: undefined,
  disabled: false,
};

const ActiveValidationIcon = () => (
  <Icon className="validation__message__icon--active" name="close-small" size={12} />
);
const InactiveValidationIcon = () => (
  <Icon className="validation__message__icon--inactive" name="check-small" size={14} />
);
const UnsetValidationIcon = () => <div className="validation__message__icon--unset" />;

const ValidationIcon = ({ active }) => {
  if (active === true) {
    return <ActiveValidationIcon />;
  } else if (active === false) {
    return <InactiveValidationIcon />;
  }
  return <UnsetValidationIcon />;
};

const ValidationMessage = ({ message, active }) => (
  <Row>
    <li
      className={`validation__message ${active === false ? 'validation__message--inactive' : ''}`}
    >
      <div className="validation__message__icon-container">
        <ValidationIcon active={active} />
      </div>
      <span className="validation__message__text">{message}</span>
    </li>
  </Row>
);

ValidationMessage.propTypes = {
  message: PropTypes.string,
  active: PropTypes.bool,
};
ValidationMessage.defaultProps = {
  message: undefined,
  active: undefined,
};

const ValidationMessages = ({ messages }) => {
  let validationMessageList = null;
  if (messages.length) {
    validationMessageList = messages.map(({ message, active }, i) => (
      <ValidationMessage key={i.toString()} message={message} active={active} />
    ));
  }
  return <ul className="validation__messages">{validationMessageList}</ul>;
};

ValidationMessages.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(ValidationMessage.propTypes)),
};

ValidationMessages.defaultProps = {
  messages: [],
};

const InvalidIcon = ({ size }) => (
  <Icon
    size={iconSizes[size]}
    name="warning-sign"
    className="mb-input__inner-icon validation__icon validation__icon--invalid "
    tooltip="Invalid"
    tooltipDelay={500}
    tooltipKind="error"
  />
);

const ValidationWrapper = ({ messages, active, children }) => (
  <Tooltip
    position="right"
    kind="neutral"
    custom
    delay={500}
    content={<ValidationMessages messages={messages} />}
    forceVisibility={active === true && messages && messages.length > 0}
    showOnHover={false}
  >
    {children}
  </Tooltip>
);

const NamedIcon = ({ namedIconTitle }) => <div className="named__icon">{namedIconTitle[0]}</div>;

NamedIcon.propTypes = {
  namedIconTitle: PropTypes.string,
};
NamedIcon.defaultProps = {
  namedIconTitle: undefined,
};

export { Loader, Placeholder, InnerButton, ValidationWrapper, InvalidIcon, NamedIcon };
