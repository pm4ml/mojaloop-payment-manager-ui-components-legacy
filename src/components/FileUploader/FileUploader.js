import '../../icons/mule/upload-small.svg';
import '../../icons/mule/close-small.svg';
import './FileUploader.scss';

import isNil from 'lodash/isNil';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import keyCodes from '../../utils/keyCodes';
import { InnerButton, InvalidIcon, Loader, Placeholder, ValidationWrapper } from '../Common';

class FileUploader extends PureComponent {
  constructor(props) {
    super(props);

    this.onClickFileUploader = this.onClickFileUploader.bind(this);
    this.onCloseFileUploader = this.onCloseFileUploader.bind(this);
    this.leaveFileUploader = this.leaveFileUploader.bind(this);
    this.onEnterFileUploader = this.onEnterFileUploader.bind(this);
    this.onPageClick = this.onPageClick.bind(this);
    this.onButtonClick = this.onButtonClick.bind(this);
    this.onRemoveButtonClick = this.onRemoveButtonClick.bind(this);
    this.onChangeFile = this.onChangeFile.bind(this);
    this.onKeyDown = this.onKeyDown.bind(this);

    this.state = {
      isOpen: false,
      fileName: this.props.fileName,
      fileContent: this.props.file,
    };
  }

  componentDidMount() {
    window.addEventListener('mouseup', this.onPageClick, false);
  }
  componentDidUpdate(prevProps) {
    const { file, disabled } = this.props;
    if (file && file !== this.state.fileContent) {
      this.setState({ fileContent: file });
    } else if (!file && file !== prevProps.file) {
      this.setState({
        fileContent: undefined,
        fileName: undefined,
      });
    }
    if (disabled !== prevProps.disabled) {
      this.setState({ isOpen: false });
    }
  }
  componentWillUnmount() {
    window.removeEventListener('mouseup', this.onPageClick, false);
  }
  async onChangeFile(e) {
    const readAsText = file => {
      const reader = new FileReader();
      return new Promise((resolve, reject) => {
        reader.onload = event => resolve(event.target.result);
        reader.onerror = error => reject(error);
        reader.readAsText(file);
      });
    };
    const readAsBase64 = file => {
      const reader = new FileReader();
      return new Promise((resolve, reject) => {
        reader.readAsDataURL(file);
        reader.onload = event => resolve(event.target.result.split(',')[1]);
        reader.onerror = error => reject(error);
      });
    };

    const [file] = e.target.files;
    if (file === undefined) {
      return;
    }

    const { parseFileAs } = this.props;
    let parseFile = file;
    if (parseFileAs === 'text') {
      parseFile = await readAsText(file);
    }
    if (parseFileAs === 'base64') {
      parseFile = await readAsBase64(file);
    }
    this.setState({
      fileContent: file,
      fileName: file.name,
    });

    if (this.props.onChange) {
      this.props.onChange(parseFile);
    }
  }
  onCloseFileUploader() {
    this.setState({ isOpen: false });
  }
  onEnterFileUploader() {
    this.setState({ isOpen: true });
    if (this.state.disabled) {
      this.leaveFileUploader();
    }
  }
  onClickFileUploader() {
    const isOpen = !this.state.isOpen;
    this.setState({ isOpen }, () => {
      if (isOpen === true) {
        this.fileuploader.focus();
      }
    });
  }
  onButtonClick() {
    this.fileuploader.click();
  }
  onRemoveButtonClick() {
    this.fileuploader.value = '';
    this.fileContent = undefined;
    this.setState({
      fileName: undefined,
    });
    if (this.props.onChange) {
      this.props.onChange(undefined);
    }
  }

  onKeyDown(e) {
    if (!this.state.isOpen) {
      return;
    }

    if (e.nativeEvent.keyCode === keyCodes.KEY_SHIFT) {
      // Handle an issue where fileupload loses focus when pressing the Shift Key
      e.preventDefault();
      e.stopPropagation();
      return;
    }
    const { keyCode, shiftKey } = e.nativeEvent;
    if (keyCode === keyCodes.KEY_TAB) {
      e.preventDefault();
      e.stopPropagation();
      this.leaveFileUploader(!shiftKey);
      return;
    }

    if (keyCode === keyCodes.KEY_RETURN) {
      e.preventDefault();
      if (isNil(this.fileContent)) {
        this.onButtonClick();
      } else {
        this.onRemoveButtonClick();
      }
    }
  }
  onPageClick(e) {
    if (!this.state.isOpen) {
      return;
    }
    const isClickWithinTextFieldBox = this.area.contains(e.target);
    if (!isClickWithinTextFieldBox) {
      this.onCloseFileUploader();
    }
  }
  leaveFileUploader(next = true) {
    utils.focusNextFocusableElement(this.fileuploader, next);
    this.onCloseFileUploader();
  }

  render() {
    const {
      id,
      size,
      className,
      placeholder,
      fileType,
      style,
      required,
      invalid,
      invalidMessages,
      pending,
      disabled,
    } = this.props;
    const { isOpen, fileName } = this.state;
    const hasFile = !isNil(this.state.fileContent) && fileName;

    const componentClassName = utils.composeClassNames([
      className,
      'input-fileuploader__component',
      'mb-input',
      'mb-input__borders',
      'mb-input__background',
      'mb-input__shadow',
      size === 's' && 'mb-input--small',
      size === 'm' && 'mb-input--medium',
      size === 'l' && 'mb-input--large',
      /* eslint-disable max-len  */
      isOpen &&
        'mb-input--open mb-input__borders--open mb-input__background--open mb-input__shadow--open',
      disabled && 'mb-input--disabled mb-input__borders--disabled mb-input__background--disabled',
      pending &&
        'mb-input--pending mb-input__borders--pending mb-input__background--pending mb-input__shadow--pending',
      invalid &&
        'mb-input--invalid mb-input__borders--invalid mb-input__background--invalid mb-input__shadow--invalid',
      required &&
        !hasFile &&
        'mb-input--required mb-input__borders--required mb-input__background--required mb-input__shadow--required',
      /* eslint-enable */
    ]);

    let customPlaceholder = null;
    if (placeholder) {
      customPlaceholder = <Placeholder size={size} label={placeholder} active />;
    }

    const fileNameClassName = utils.composeClassNames([
      'input-fileuploader__filename',
      !hasFile && 'input-fileuploader__filename--no-file',
      size === 's' && 'input-fileuploader__filename--small',
      size === 'm' && 'input-fileuploader__filename--medium',
      size === 'l' && 'input-fileuploader__filename--large',
    ]);
    const fileNameLabel = hasFile ? fileName : 'No File Chosen';

    let fileButton = null;
    if (!pending) {
      let buttonClassName = 'input-fileuploader__button-add';
      let onClick = this.onButtonClick;
      let label = 'Choose File';
      let icon = 'upload-small';
      let kind = isOpen ? 'primary' : 'tertiary';

      if (hasFile) {
        buttonClassName = 'input-fileuploader__button-remove';
        onClick = this.onRemoveButtonClick;
        label = 'Remove File';
        icon = 'close-small';
        if (isOpen) {
          kind = 'danger';
        }
      }
      fileButton = (
        <InnerButton
          size={size}
          tabIndex="-1"
          className={buttonClassName}
          onClick={onClick}
          kind={isOpen && required && !hasFile ? 'warning' : kind}
          noFill={!isOpen}
          icon={icon}
          label={label}
          disabled={disabled}
          active={isOpen}
        />
      );
    }

    let invalidIcon = null;
    if (invalid) {
      invalidIcon = <InvalidIcon size={size} />;
    }

    let loader = null;
    if (pending) {
      loader = <Loader size={size} />;
    }

    return (
      <ValidationWrapper messages={invalidMessages} active={isOpen}>
        <div
          style={style}
          id={id}
          className={componentClassName}
          onClick={this.onClickFileUploader}
          onKeyDown={this.onClickFileUploader}
          ref={area => {
            this.area = area;
          }}
          role="presentation"
        >
          <div className="input-fileuploader-box">
            <div className="mb-input__content input-fileuploader__content">
              {customPlaceholder}
              <input
                className="input-fileuploader__input"
                type="file"
                accept={fileType}
                onFocus={this.onEnterFileUploader}
                onChange={this.onChangeFile}
                disabled={disabled}
                ref={fileuploader => {
                  this.fileuploader = fileuploader;
                }}
                onKeyDown={this.onKeyDown}
                id={`${id}-file`}
              />
              <div className={fileNameClassName}>{fileNameLabel}</div>
              {loader}
              {invalidIcon}
              {fileButton}
            </div>
          </div>
        </div>
      </ValidationWrapper>
    );
  }
}

FileUploader.propTypes = {
  style: PropTypes.shape(),
  id: PropTypes.string,
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  file: PropTypes.string,
  fileName: PropTypes.string,
  fileType: PropTypes.string,
  parseFileAs: PropTypes.oneOf(['text', 'base64']),
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  pending: PropTypes.bool,
  disabled: PropTypes.bool,
  invalid: PropTypes.bool,
  invalidMessages: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      text: PropTypes.string,
    }),
  ),

  required: PropTypes.bool,
};

FileUploader.defaultProps = {
  style: {},
  id: undefined,
  className: undefined,
  size: 'l',
  file: undefined,
  fileName: undefined,
  fileType: undefined,
  parseFileAs: undefined,
  onChange: undefined,
  placeholder: undefined,
  pending: false,
  disabled: false,
  invalid: false,
  invalidMessages: [],
  required: false,
};

export default FileUploader;
