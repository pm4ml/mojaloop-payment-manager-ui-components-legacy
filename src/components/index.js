import '../assets/styles/index.scss';

import Button from './Button';
import Checkbox from './Checkbox';
import Column from './Column';
import ContentReader from './ContentReader';
import ControlIcon from './ControlIcon';
import DataList, { Link } from './DataList';
import DatePicker from './DatePicker';
import Dropdown from './Dropdown';
import FileUploader from './FileUploader';
import FormInput, { FormInputs, FormInputsGroup, Inline } from './FormInputs';
import Heading from './Heading';
import Icon from './Icon';
import Menu, { MenuItem, MenuSection } from './Menu';
import MessageBox from './MessageBox';
import Modal, { ModalTabsLayout } from './Modal';
import Navbar from './Navbar';
import Pill from './Pill';
import RadioGroup from './RadioGroup';
import Row from './Row';
import ScrollBox from './ScrollBox';
import Select from './Select';
import Spinner from './Spinner';
import { Tab, TabList, TabPanel, TabPanels, Tabs } from './Tabs';
import TextArea from './TextArea';
import TextField from './TextField';
import Toast from './Toast';
import Toggle from './Toggle';
import Tooltip from './Tooltip';

export {
  Button,
  Column,
  ContentReader,
  ControlIcon,
  Checkbox,
  DataList,
  DatePicker,
  Dropdown,
  FileUploader,
  FormInput,
  FormInputs,
  FormInputsGroup,
  Heading,
  Icon,
  Inline,
  Link,
  Menu,
  MenuSection,
  MenuItem,
  MessageBox,
  Modal,
  ModalTabsLayout,
  Navbar,
  Pill,
  RadioGroup,
  Row,
  ScrollBox,
  Select,
  Spinner,
  Tabs,
  Tab,
  TabList,
  TabPanels,
  TabPanel,
  TextArea,
  TextField,
  Toast,
  Toggle,
  Tooltip,
};
