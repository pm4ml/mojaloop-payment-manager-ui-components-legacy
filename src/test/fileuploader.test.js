import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { InvalidIcon, Loader, Placeholder, ValidationWrapper } from '../components/Common';
import FileUploader from '../components/FileUploader';

it('renders the fileuploader', () => {
  const wrapper = shallow(<FileUploader />);
  expect(wrapper.find('div.input-fileuploader__component')).toHaveLength(1);
});

it('renders the filuploader value', () => {
  const wrapper = shallow(<FileUploader />);
  expect(wrapper.find('.input-fileuploader__filename').text()).toBe('No File Chosen');
});

it('renders the placeholder', () => {
  const wrapper = shallow(<FileUploader placeholder="test-fileuploader" />);
  expect(wrapper.find(Placeholder)).toHaveLength(1);
  expect(wrapper.find(Placeholder).prop('label')).toBe('test-fileuploader');
});

it('renders the validation wrapper', () => {
  const wrapper = shallow(<FileUploader placeholder="test-Select" />);
  expect(wrapper.find(ValidationWrapper)).toHaveLength(1);
});

it('applies the prop className', () => {
  const wrapper = shallow(<FileUploader className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(<FileUploader id="testFileUploaderId" />);
  expect(wrapper.find('#testFileUploaderId')).toHaveLength(1);
});

it('renders the disabled state', () => {
  const wrapper = shallow(<FileUploader disabled />);
  expect(wrapper.find('input')).toHaveLength(1);
  expect(wrapper.find('input').prop('disabled')).toBe(true);
});

it('renders the pending state', () => {
  const wrapper = shallow(<FileUploader pending />);
  expect(wrapper.find(Loader)).toHaveLength(1);
});

it('renders the invalid state', () => {
  const wrapper = shallow(<FileUploader invalid />);
  expect(wrapper.find(InvalidIcon)).toHaveLength(1);
});

it('renders the required state', () => {
  const wrapper = shallow(<FileUploader required />);
  expect(wrapper.find('.mb-input--required')).toHaveLength(1);
});

it('renders the large, medium, small sizes', () => {
  const sizes = {
    large: 'l',
    medium: 'm',
    small: 's',
  };
  Object.entries(sizes).forEach(([name, size]) => {
    const wrapper = shallow(<FileUploader size={size} />);
    const className = `mb-input--${name}`;
    expect(wrapper.find('.mb-input').hasClass(className)).toBeTruthy();
  });
});

// Snapshot

it('renders the fileuploader correctly when multiple props are set', () => {
  const wrapper = shallow(<FileUploader id="test-id" />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
