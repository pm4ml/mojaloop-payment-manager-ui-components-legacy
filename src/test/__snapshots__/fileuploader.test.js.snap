// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the fileuploader correctly when multiple props are set 1`] = `
<ValidationWrapper
  active={false}
  messages={Array []}
>
  <div
    className="input-fileuploader__component mb-input mb-input__borders mb-input__background mb-input__shadow mb-input--large"
    id="test-id"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    style={Object {}}
  >
    <div
      className="input-fileuploader-box"
    >
      <div
        className="mb-input__content input-fileuploader__content"
      >
        <input
          className="input-fileuploader__input"
          disabled={false}
          id="test-id-file"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="file"
        />
        <div
          className="input-fileuploader__filename input-fileuploader__filename--no-file input-fileuploader__filename--large"
        >
          No File Chosen
        </div>
        <InnerButton
          active={false}
          className="input-fileuploader__button-add"
          disabled={false}
          icon="upload-small"
          kind="tertiary"
          label="Choose File"
          noFill={true}
          onClick={[Function]}
          size="l"
          tabIndex="-1"
        />
      </div>
    </div>
  </div>
</ValidationWrapper>
`;
